tests/federation_room_join_partial_state_test.go

```diff
@@ -642,10 +645,17 @@ func beginPartialStateJoin(t *testing.T, deployment *docker.Deployment, joiningU
 
 	// register a handler for /state_ids requests, which finishes fedStateIdsRequestReceivedWaiter, then
 	// waits for fedStateIdsSendResponseWaiter and sends a reply
-	handleStateIdsRequests(t, result.Server, result.ServerRoom, result.fedStateIdsRequestReceivedWaiter, result.fedStateIdsSendResponseWaiter)
+	handleStateIdsRequests(
+		t,
+		result.Server,
+		result.ServerRoom,
+		result.fedStateIdsRequestReceivedWaiter,
+		result.fedStateIdsSendResponseWaiter,
+		result.fedStateIdsAllowedEvents,
```

Comment:
I think it might be easier to flip this around, and have handleStateIdsRequests only match /state_ids requests for the last event before the join event? That's what I did in #418, in https://github.com/matrix-org/complement/pull/418/commits/687ad360910260a0cf6167cf3df36a97694297b4.

Reply:
That looks a lot cleaner. I've borrowed that commit and it works great. Do we want to split that change into a separate PR?

Reply:
yeah, I'll do that.

Reply:
#421